src/main/java/org/mondemand/Client.java

```diff
@@ -549,6 +558,34 @@ public void increment (StatType type, String key, int value) {
   }
 
   /**
+   * Given context&Stats map, increment according to context and key/value
+   * @param contextStats ConcurrentHashMap<Context, AtomicLongMap<String>> contextStats
+   * @param context context
+   * @param keyType key
+   * @param value value
+   */
+  public synchronized void increment(ContextList context, String keyType, long value)
+  {
+    AtomicLongMap<String> stats = contextStats.get(context);
+    if (stats == null)
+    {
+      stats = AtomicLongMap.create();
+      contextStats.put(context, stats);
```

Comment:
Change these lines to the following and the unit tests will pass without synchronizing this method.
````
    contextStats.putIfAbsent(context, stats);
    stats = contextStats.get(context);
````